src/asmcnc/production/database/factory_payload_sender.py

```diff
@@ -0,0 +1,68 @@
+import os
+import csv
+import uuid
+
+CSV_PATH = 'asmcnc/production/database/csvs/'
+WORKING_DIR = 'A:\\factory_data\\data\\new\\'
+
+
+def get_unique_file_name(machine_serial, table, stage):
+    return str(machine_serial) + '-' + str(table) + '-' + str(stage) + '-' + str(uuid.uuid4()) + '.csv'
+
+
+def get_csv(json, machine_serial, table, stage=None, csv_path=CSV_PATH):
+    if csv_path is None:
+        csv_path = CSV_PATH
+
+    if not os.path.exists(csv_path):
+        os.mkdir(csv_path)
+
+    file_path = csv_path + get_unique_file_name(machine_serial, table, stage)
+
+    keys = json[0].keys()
+
+    with open(file_path, 'w') as data_file:
+        dict_writer = csv.DictWriter(data_file, keys)
+        dict_writer.writeheader()
+        dict_writer.writerows(json)
+
+    return file_path
+
+
+def send_csv_to_ftp(file_path, ftp_server=None, ftp_username=None, ftp_password=None):
```

Comment:
there are a few things going on in this function: 
-  setting up the credentials file
- opening and closing ssh
- opening and closing file transfer
- doing the file transfer


for readability and testing it might be worth considering refactoring it into more functions in future.